core/ecschema-editing/src/Differencing/SchemaDifference.ts

```diff
@@ -102,119 +119,260 @@ export interface SchemaDifferences {
   readonly sourceSchemaName: string;
   readonly targetSchemaName: string;
 
-  readonly changes: AnySchemaDifference[];
-  readonly conflicts: SchemaDifferenceConflict[];
+  readonly changes?: AnySchemaDifference[];
+  readonly conflicts?: SchemaDifferenceConflict[];
 }
 
 /**
  * @internal
  */
 export type AnySchemaDifference =
```

Comment:
how much value do we get by treating absolutely every difference polymorphicly?  Do we need to do such complex type magic to get that value?